pkg/tfbridge/detailed_diff.go

```diff
@@ -259,6 +292,78 @@ func (differ detailedDiffer) makeListDiff(
 	return diff
 }
 
+type setChangeIndex struct {
+	engineIndex   int
+	newStateIndex int
+	oldChanged    bool
+	newChanged    bool
+}
+
+func (differ detailedDiffer) makeSetDiff(
+	path propertyPath, old, new resource.PropertyValue,
+) map[detailedDiffKey]*pulumirpc.PropertyDiff {
+	diff := make(map[detailedDiffKey]*pulumirpc.PropertyDiff)
+	oldList := old.ArrayValue()
+	newList := new.ArrayValue()
+	newInputsList := []resource.PropertyValue{}
+
+	newInputs, newInputsOk := path.GetFromMap(differ.newInputs)
+	if newInputsOk && isPresent(newInputs) && newInputs.IsArray() {
+		newInputsList = newInputs.ArrayValue()
+	}
+
+	oldIdentities := differ.calculateSetHashIndexMap(path, resource.NewArrayProperty(oldList))
+	newIdentities := differ.calculateSetHashIndexMap(path, resource.NewArrayProperty(newList))
+	inputIdentities := hashIndexMap{}
+
+	if !schemaContainsComputed(path, differ.tfs, differ.ps) {
+		// The inputs are only safe to hash if the schema has no computed properties
+		inputIdentities = differ.calculateSetHashIndexMap(path, resource.NewArrayProperty(newInputsList))
+	}
+
+	// The old indices and new inputs are the indices the engine can reference
+	// The new state indices need to be translated to new input indices when presenting the diff
+	setIndices := make(map[int]setChangeIndex)
+	for hash, oldIndex := range oldIdentities {
+		if _, newOk := newIdentities[hash]; !newOk {
+			setIndices[oldIndex] = setChangeIndex{engineIndex: oldIndex, oldChanged: true, newStateIndex: -1, newChanged: false}
+		}
+	}
+	for hash, newIndex := range newIdentities {
+		if _, oldOk := oldIdentities[hash]; !oldOk {
+			inputIndex := inputIdentities[hash]
+			if inputIndex == -1 {
+				glog.Warningln(
+					"Element at path %s in new state not found in inputs, the displayed diff might be inaccurate",
+					path.String())
```

Comment:
Can we use `GetLogger(ctx)` here? That way the warning would be associated with the correct resource.
```suggestion
				GetLogger(ctx).Warn(fmt.Sprintf(
					"Element at path %s in new state not found in inputs, the displayed diff might be inaccurate",
					path.String()))
```